float64 new_x
float64 new_y
float64 new_th
---
float64 old_x
float64 old_y
float64 old_th